src/AppBundle/Entity/Report/Traits/ReportProfDeputyCostsTrait.php

```diff
@@ -199,19 +199,17 @@ public function profCostsHowChangedAtLeastOne(ExecutionContextInterface $context
      */
     public function profCostsInterimAtLeastOne(ExecutionContextInterface $context)
     {
-        // skip validation if there is at least one interim with date
-        foreach($this->getProfDeputyInterimCosts() as $ic) {
-            if (!empty($ic->getAmount()) && !empty($ic->getDate())) {
-                return;
-            }
+        $ics = $this->getProfDeputyInterimCosts();
+
+        if ($ics[0]->getDate() === null) {
+            $context->buildViolation('profDeputyInterimCost.date.notBlank')->atPath('profDeputyInterimCosts[0].date')->addViolation();
```

Comment:
what happens if there are no Interim costs? Wont we get an undefined index at 0 ?

Reply:
Yep good catch - the way the validation checks were running was hiding this during local testing as the `NotBlank()` on the entity was being run first but if we ever change the order of validation this would become a problem so I've added a `!empty()` conditional around them.